packages/flame/lib/src/game/game.dart

```diff
@@ -202,6 +202,7 @@ abstract mixin class Game {
   ///
   /// Should not be called manually.
   void detach() {
+    onRemove();
     _gameRenderBox = null;
 
     onDetach();
```

Comment:
Sorry, can you remind me again why this is not done in `onDetech` instead of `onRemove`?

Reply:
Sorry for taking so long to respond to all of this.  I have / am still out on work travel and have been slammed.  The issue is that components had onRemove that got called BUT game.dart's onRemove never gets called - ever.  So, the concept of the game level event method BEFORE it is removed from the tree, even though it says it does this in the docs, it never makes that call.  This gets the onRemove method called for game now before it is detached.  This needs to be called when game is actually being detached.  I created this graphic just now, because if you chase the events all the way up into the Flutter widget tree, you will see this event was being missed by Flame.

![flame_tree](https://github.com/flame-engine/flame/assets/17042581/2e786516-903b-4226-bdb6-cafe42d1a967)